libs/redis/langchain_redis/vectorstores.py

```diff
@@ -933,200 +1011,78 @@ def similarity_search_with_score_by_vector(
 
             if with_vectors:
                 return_fields.append(self.config.embedding_field)
-        else:
-            return_fields = []
 
         if distance_threshold is None:
```

Comment:
Could be worth to look at a `_build_query(...)` helper method that takes inputs and spits out a redisvl query since we do this more than 1x

Reply:
I see this in redisvl https://github.com/redis/redis-vl-python/blob/611e1b296648a240f8309d0eb4091fa7e5842acc/redisvl/query/query.py#L350-L351

but I'm not sure about the intended use here might need you to provide the upshot

Reply:
Ahh, I simply mean add a class method to the vectorstore here where you wrap the logic for creating the `VectorQuery` or `RangeQuery` objects (by checking distance threshold) since it's duplicated in two different methods

Reply:
ahh gotcha